# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/MediaTrackListener.cpp
# Commit: 35d86243cb45
# Full Hash: 35d86243cb45f7e02dd6ec0eea04cd62c2a70196
# Author: Andrew Osmond <aosmond@mozilla.com>
# Date: 2024-09-16 09:14:26
# Regressor Bug: 1908245
# File Overlap Count: 1
# Description:
#   Bug 1908245 - Add support for missing requestVideoFrameCallback parameters. r=media-playback-reviewers,webidl,padenot,saschanaz
#   
#   This patch adds support for exposing the capture time, receive time,
#   processing duration and RTP timestamp parameters on the callback for
#   rVFC for WebRTC related elements.
# ==============================================================================

diff -r 7d143d92bf2a -r 35d86243cb45 dom/media/MediaTrackListener.cpp
--- a/dom/media/MediaTrackListener.cpp	Sun Sep 15 20:03:40 2024 +0000
+++ b/dom/media/MediaTrackListener.cpp	Sun Sep 15 21:49:38 2024 +0000
@@ -25,9 +25,7 @@
     VideoSegment& aFrom, VideoSegment& aTo, DisabledTrackMode aMode) {
   if (aMode == DisabledTrackMode::SILENCE_BLACK) {
     for (VideoSegment::ChunkIterator it(aFrom); !it.IsEnded(); it.Next()) {
-      aTo.AppendFrame(do_AddRef(it->mFrame.GetImage()),
-                      it->mFrame.GetIntrinsicSize(), it->GetPrincipalHandle(),
-                      true);
+      aTo.AppendFrame(*it, Some(true));
       aTo.ExtendLastFrameBy(it->GetDuration());
     }
   } else if (aMode == DisabledTrackMode::SILENCE_FREEZE) {